common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -1157,32 +1160,61 @@
          */
         Problem.prototype.disableAllButtonsWhileRunning = function(operationCallback, isFromCheckOperation) {
             var that = this;
-            this.enableAllButtons(false, isFromCheckOperation);
+            var allButtons = [this.resetButton, this.saveButton, this.showButton, this.hintButton, this.submitButton];
+            var disabledButtons = this.disableButtons(allButtons, isFromCheckOperation);
```

Comment:
Part of my confusion here is that these aren't all the disabledButtons, they are just the ones that the disableButtons method operated on. I'm trying to think of the clearest way to do this.

It seems there are 2 options-- try to come up with clearer variable names (both here and in disableButtons), or filter "allButtons" to only have the enabled buttons prior to calling disableButtons (in which case disableButtons doesn't need to return anything). I'll let you decide your preference.